runtime/src/bank.rs

```diff
@@ -4688,6 +4707,35 @@ impl Bank {
                         program_runtime_environment_v1.clone(),
                     )
                 }),
+
+            ProgramAccountLoadResult::ProgramOfLoaderV4(program_account, slot) => {
+                let loaded_program = program_account
+                    .data()
+                    .get(LoaderV4State::program_data_offset()..)
+                    .and_then(|elf_bytes| {
+                        LoadedProgram::new(
+                            &loader_v4::id(),
+                            program_runtime_environment_v2.clone(),
+                            slot,
+                            slot.saturating_add(DELAY_VISIBILITY_SLOT_OFFSET),
```

Comment:
Why delay visibility here? This is not a deployment, right?

Reply:
The `slot` is the deployment slot. So the effective slot is `+ DELAY_VISIBILITY_SLOT_OFFSET`.